CMakeLists.txt

```diff
@@ -64,13 +64,31 @@ set(NMC_THREADING_MODEL "cthread" CACHE STRING "set the threading model, one of
 set_property(CACHE NMC_THREADING_MODEL PROPERTY STRINGS cthread tbb serial )
 
 if(NMC_THREADING_MODEL MATCHES "tbb")
+    # download and build TBB
+    include(cmake/TBBGet.cmake)
+    include(cmake/TBBBuild.cmake)
+    tbb_get(TBB_ROOT tbb_root SOURCE_CODE)
+    tbb_build(TBB_ROOT ${tbb_root} CONFIG_DIR TBB_DIR)
```

Comment:
I am concerned that this requires that the build environment has access to the wider internet, which historically is not necessarily the case in some hpc environments.